### PR TITLE
jquery__pointernative: pointerenter / pointerleave events bubble up to the document root

### LINKED ISSUE #801: jquery__pointernative: pointerenter / pointerleave events bubble up to the document root
As `pointerenter` and `pointerleave` are equivalents of native `mouseenter` and `mouseleave` events respectively, they both must not bubble up through the DOM. 

See [`mouseleave` event description](https://developer.mozilla.org/en-US/docs/Web/Events/mouseleave) for reference.

### PR DESCRIPTION
**Pull request by narqo**
fix #801

/cc @dfilatov

### CODE REVIEW

**Comment by narqo** [comment]
Note, the only browsers I've tested it are Chrome and Fx.

**Comment by dfilatov** [comment]
Could you add spec for `pointerleave`?

**Comment by narqo** [comment]
@dfilatov we already had simple tests for `pointerleave / pointerleave` events. But we didn't test event bubbling. 

:up:

**Comment by narqo** [comment]
BTW: It seems that no one of currently popular Pointer Events Polyfils (MS, Polymer or jquery's PEP) would pass [this test case](https://github.com/bem/bem-core/pull/802/files#diff-810fc4229b7cb8db80e663abf226a289R160), as they use native `bubbling = false` and dispatch the event on the last nested target only  :metal:

File: common.blocks/jquery/__event/_type/jquery__event_type_pointernative.js [hunk 1]
```diff
@@ -258,15 +268,23 @@ var dispatcher = {
...
-        if(!this.contains(event.target, event.relatedTarget)) {
-            this.enter(event);
+        this.enterLeave(event, this.enter);
+    },
+
+    enterLeave : function(event, fn) {
+        var target = event.target,
+            relatedTarget = event.relatedTarget;
+
+        while(target && !this.contains(target, relatedTarget)) {
```
Comment (dfilatov) [inline common.blocks/jquery/__event/_type/jquery__event_type_pointernative.js]: Isn't it expensive? `contains` has inner cycle as well

Comment (narqo) [inline common.blocks/jquery/__event/_type/jquery__event_type_pointernative.js]: Hm, yep, haven't thought about it :( 

What if I rewrite it like this:

``` js
if(!this.contains(target, relatedTarget)) {
  do {
    event.target = target;
    fn.call(this, event);  // we need to emit non bubbling events here on the subtree from target to relatedTarget
  } while(target = target.parentNode && target !== relatedTarget)
}
```

?

Comment (dfilatov) [inline common.blocks/jquery/__event/_type/jquery__event_type_pointernative.js]: imho looks better

**Comment by narqo** [comment]
:up:

File: common.blocks/jquery/__event/_type/jquery__event_type_pointernative.js [hunk 2]
```diff
@@ -258,15 +268,25 @@ var dispatcher = {
...
     },
 
     enterOver : function(event) {
         this.over(event);
-        if(!this.contains(event.target, event.relatedTarget)) {
-            this.enter(event);
+        this.enterLeave(event, this.enter);
+    },
+
+    enterLeave : function(event, fn) {
```
Comment (narqo) [inline common.blocks/jquery/__event/_type/jquery__event_type_pointernative.js:279]: I don't like the name of the method, but I couldn't decide how to call it :(

**Comment by dfilatov** [comment]
:+1:

**Comment by dfilatov** [comment]
@narqo Don't forget do the same PR to v3

**Pull request merged**